web/src/hooks/usePagination.ts

```diff
@@ -30,5 +33,14 @@ export default function usePagination() {
     history.replace(`${location.pathname}?page=${page}&pageSize=${pageSize}`);
   };
 
-  return { paginationConfig, savePageList };
+  const checkPageList = (ref: MutableRefObject<ActionType | undefined>) => {
+    const { current, pageSize, total } = ref.current!.pageInfo as PageInfo;
+    if (current > pageSize / total && current > 1) {
+      savePageList(paginationConfig.current - 1, paginationConfig.pageSize);
+    } else {
+      ref.current?.reload();
+    }
+  };
+
+  return { paginationConfig, savePageList, checkPageList };
```

Comment:
Since we have abstracted the pager into a hook, is it enough for us to write a test for the pagination